src/importexport/musicxml/internal/musicxml/export/exportmusicxml.cpp

```diff
@@ -6993,6 +6989,13 @@ void ExportMusicXml::keysigTimesig(const Measure* m, const Part* p)
                 }
             }
         }
+        for (staff_idx_t i = 1; i < nstaves; i++) {
+            // return here if it's a key change because of an instrument change
+            if (keysigs.at(i)->forInstrumentChange()) {
```

Comment:
This might be problematic in the edge case that there is an instrument change with a corresponding key signature change on one staff, and a non-instrument-change-related key signature on another staff

Reply:
Highly unlikely to me, but better to be sure anyway.